### ISSUE TITLE
Create an annotation to ignore parameters for key generation

### ISSUE THREAD

**Issue by mdedetrich** [cb372/scalacache#30]
Note sure if this is possible, but it would be fantastic if scala cache provided an annotation (lets say `@ignoreCache`) which would allow you to tell the memoize function to ignore certain parameters when it generates a key.

It would be a lot less verbose than having to create a `MethodCallToStringConverter`, and its a very common use case we have (i.e. we have database connections as parameters, which we really don't want to be part of the cache key, and writing a seperate implementation of `MethodCallToStringConverter` for every single method gets tedious ).

Was thinking of something along these lines

```scala
def doSomething(userId:UserId)(implicit @ignoreCache databaseConnect:DatabaseConnection) = memoize {..}
```

The `@ignoreCache` in this case would mean that the key for doSomething will only depend on the `userId`.

**Comment by cb372** [comment]
That's a good idea.

In general I'm not a huge fan of annotations, but in this case it seems like a valid use case for them. I'll do some experimentation about the best way to implement this, either using annotations or otherwise.

**Comment by cb372** [comment]
Done! Implementation was easier than I expected.

Usage is exactly as you suggested, but I decided to call the annotation `@cacheKeyExclude`.

I've just released 0.5.2, should hit Maven Central in a couple of hours. Let me know if you have any problems.

**Comment by mdedetrich** [comment]
Cool thanks, will check it out in a bit (unfortunately can't use `0.5.2` straight away because we are stuck with using a certain version of the `scala-logging` library, so need to fix this first on our end first)
